services/blockstorage/sync/state_idle_test.go

```diff
@@ -0,0 +1,45 @@
+package sync
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestIdleStateStaysIdleOnCommit(t *testing.T) {
+	h := newBlockSyncHarness()
+	idle := h.sf.CreateIdleState()
+	var next syncState = nil
+	latch := make(chan struct{})
+	go func() {
+		next = idle.processState(h.ctx)
+		latch <- struct{}{}
+	}()
+	idle.blockCommitted()
+	<-latch
+	require.True(t, next != idle, "processState state should be a different idle state (which was restarted)")
```

Comment:
you're actually not checking that the next state *is* idle; should add a `require.IsType` before.

Reply:
correct, but this entire test is wrong, i added what was missing, but i think it should be rewritten as it assumes that the implementation will return a new state - maybe it should just commit some blocks in a rate faster than the timeout and then check that the state is still idle